### PR TITLE
Paypal components

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

> Components to provide end users with details about their PayPal payouts made through the new PayPal integration.

> initial new components: 
 - \<sqp-account-details>
 - \<sqp-status-column>
 - \<sqp-rewards-column>

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

- Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3345
- Process.st launch checklist: https://app.process.st/runs/PayPal-Components-q5l0eCl1I56kipYI9EVJDQ

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run (if applicable)
- [x] The behaviour changes in the pull request are covered by specs
- [x] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist

### Code review

- [x] Changes have been reviewed by at least one other engineer
- [x] Security impacts of this change have been considered

### CODE REVIEW

File: packages/paypal-components/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "allowUnreachableCode": false,
+    "declaration": false,
+    "experimentalDecorators": true,
+    "strict":false,
```
Comment (johanventer) [inline packages/paypal-components/tsconfig.json:7]: Can our components really not be built with strict?

**Review by johanventer — CHANGES_REQUESTED** [review]

File: packages/paypal-components/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
+# Changelog
```
[the remark this answers is not in the document]
Comment (DerekSiemens) [reply packages/paypal-components/CHANGELOG.md:1]: Thanks for flagging hygiene, will not launch without a proper changelog and readme

File: packages/paypal-components/src/Icons/PaypalBadge.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { h } from "@stencil/core";
+
+export const PaypalBadge = () => (
```
Comment (noahwc) [inline packages/paypal-components/src/Icons/PaypalBadge.tsx:3]: General convention has been to postfix SVGs and their files with `SVG`.

File: packages/paypal-components/src/components/sqp-account-details/AccountForm.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
+// import { h } from "@stencil/core";
```
Comment (noahwc) [inline packages/paypal-components/src/components/sqp-account-details/AccountForm.stories.tsx:1]: Calling out commented out file

File: packages/paypal-components/src/components/sqp-rewards-cell/sqp-rewards-cell.tsx [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+        "&::part(base)": {
+          background: "var(--sl-color-blue-600)",
+        },
+      },
+    };
+
+    const sheet = createStyleSheet(style);
+    const styleString = sheet.toString();
+
+    // switch (state) {
```
Comment (noahwc) [inline packages/paypal-components/src/components/sqp-rewards-cell/sqp-rewards-cell.tsx:137]: Calling out commented code

File: packages/paypal-components/src/components/sqp-rewards-column/sqp-rewards-column.tsx [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  }
+
+  @Method()
+  async renderLabel() {
+    return this.columnTitle;
+  }
+
+  @Method()
+  async renderReferrerCell(data: Referrer) {
+    // TODO: Do the right thing with many rewards, pending rewards, canceled rewards
```
Comment (noahwc) [inline packages/paypal-components/src/components/sqp-rewards-column/sqp-rewards-column.tsx:193]: Flagging TODO (and commented code above)

File: packages/paypal-components/src/components/sqp-status-cell/sqp-status-cell.tsx [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+      fontSize: "var(--sl-font-size-small)",
+      padding: "4px 8px",
+      paddingLeft: "10px",
+      whiteSpace: "pre-line",
+    },
+  },
+  RedeemBadge: {
+    "&::part(base)": {
+      fontSize: "var(--sl-font-size-small)",
+      padding: "4px 8px",
```
Comment (noahwc) [inline packages/paypal-components/src/components/sqp-status-cell/sqp-status-cell.tsx:26]: Could use CSS variables for `gap` and `padding`

File: packages/paypal-components/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+- New components
+
+  - \<sqp-account-details>
+
+    - Allows users to edit and disconnect the paypalEmail attached to their SaaSquatch user
+    - Displays a preview of any PayPal payouts that are coming in the next 3 payout periods and which currencies they will be paid out in
+    - Displays any pending rewards without a dateScheduledFor that will not be paid out by PayPal until they are available
+
+  - \<sqp-rewards-column>
+    - Used with \<sqm-referral-table> from `@saasquatch/mint-components`
```
Comment (DerekSiemens) [inline packages/paypal-components/CHANGELOG.md:23]: We should mark what version minimum is required for it to work here.

**Comment by DerekSiemens** [comment]
@00salmon Please fill in PR body

File: packages/paypal-components/src/components/sqp-status-column/sqp-status-column.tsx [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+  @Prop() pendingUnhandled: string = "Fulfillment error";
+
+  constructor() {
+    withHooks(this);
+  }
+  disconnectedCallback() {}
+
+  @Method()
+  async renderCell(data: Reward, locale: string, mintRenderer) {
+    if (mintRenderer) {
```
[the remark this answers is not in the document]
Comment (00salmon) [reply packages/paypal-components/src/components/sqp-status-column/sqp-status-column.tsx:156]: good point 👍

File: packages/paypal-components/tsconfig.json (hunk 1 above)
Comment (00salmon) [reply packages/paypal-components/tsconfig.json:7]: ts-config was copied from bedrock-components, which should also be strict 🤔 updated!

Comment (00salmon) [reply packages/paypal-components/tsconfig.json]: resolved 👍

**Review by zachharrison — DISMISSED** [review]

File: packages/paypal-components/CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [1.0.0] - 2022-11-24
```
Comment (DerekSiemens) [inline packages/paypal-components/CHANGELOG.md:10]: Flagging date change needed

File: packages/paypal-components/src/components/sqp-titled-section/sqp-titled-section.tsx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+import { PortalSectionView } from "../sqp-titled-section/sqp-portal-section-view";
+
+/**
+ * @uiName Titled Section
+ * @slots [{"name":"label", "title":"Label"},{"name":"content", "title":"Content"}]
+ * @exampleGroup Layout
+ * @example Titled Section - <sqm-titled-section text-align="center" label-margin="small" padding="none"><div slot="label"><h2>Section Label</h2></div><div slot="content"><p>Add your section content here!</p></div></sqm-titled-section>
+ */
+@Component({
+  tag: "sqm-titled-section",
```
Comment (DerekSiemens) [inline packages/paypal-components/src/components/sqp-titled-section/sqp-titled-section.tsx:14]: @00salmon noting that this folder is still in here - is there a reason? I remember we had talked about removing this

**Review by johanventer — APPROVED** [review]

**Pull request merged**